state/src/storage/tx/matching_pools.rs

```diff
@@ -0,0 +1,206 @@
+//! Helpers for accessing information about matching pools in the database
+
+use common::types::wallet::OrderIdentifier;
+use libmdbx::{TransactionKind, RW};
+
+use crate::{storage::error::StorageError, POOL_TABLE};
+
+use super::StateTxn;
+
+// -------------
+// | Constants |
+// -------------
+
+/// The name of the global matching pool
+pub const GLOBAL_MATCHING_POOL: &str = "global";
+
+/// The prefix of all matching pool keys
+pub const POOL_KEY_PREFIX: &str = "matching-pool/";
+
+/// The error message used when trying to create a pool that already exists
+const POOL_EXISTS_ERR: &str = "pool already exists";
+/// The error message used when allocating an order to a nonexistent pool
+const POOL_DOES_NOT_EXIST_ERR: &str = "pool does not exist";
+/// The error message used when a non-empty pool is attempted to be destroyed
+const POOL_NOT_EMPTY_ERR: &str = "pool not empty";
+
+// ---------------
+// | Key Helpers |
+// ---------------
+
+/// The key for the set of all matching pools
+pub fn all_pools_key() -> String {
+    "all-pools".to_string()
+}
+
+/// The key for the given order's matching pool
+pub fn matching_pool_key(order_id: &OrderIdentifier) -> String {
+    format!("{}{}", POOL_KEY_PREFIX, order_id)
+}
+
+// -----------
+// | Getters |
+// -----------
+
+impl<'db, T: TransactionKind> StateTxn<'db, T> {
```

Comment:
perhaps deferred to a follow-up, but I assume we want a getter for all the orders in a pool?

Reply:
actually, i'm not sure that we do. the intended usage is in the matching engine, where we get all of the local orders, and in the filtering applied to them [here](https://github.com/renegade-fi/renegade/blob/main/state/src/interface/order_book.rs#L181), we simply lookup the candidate order's matching pool, and see if it's equal to that of the order being matched.

the fact that that is the primary access pattern is the reason I went with a `order -> matching pool` lookup instead of the other direction